timer/timer.go

```diff
@@ -0,0 +1,120 @@
+package timer
+
+import (
+	"errors"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/nivista/steady/.gen/protos/common"
+	"github.com/nivista/steady/internal/.gen/protos/messaging"
+)
+
+type (
+	// Timer represents the configuration for the execution of a recurring task.
+	Timer struct {
+		executer  executer
+		scheduler scheduler
+	}
+
+	// Progress records the progress of a recurring task.
+	Progress struct {
+		// CompletedExecutions indicated the number of successfully completed executions.
+		CompletedExecutions int
+
+		// LastExecution indicates which execution was recorded last.
+		LastExecution time.Time
```

Comment:
This should be a pointer and should be nil if there was no last execution.

Reply:
Ok will do.